### PR TITLE
feat: introduce tagged peers and allow for libp2p, update deps

### PR DESCRIPTION
**Pull request by simone-sanfratello**
- remove deps moved to core-lib
- update deps
- remove p2p polling as it's not used anymore
- introduce "allow" list configuration
- introduce "tagged peers" configuration from Dynamo config table

### CODE REVIEW

**Comment by alanshaw** [comment]
🙏 What is the reason for loading gateway peers from dynamo?

If they change we need to restart to get the new ones tagged - right?

We load peer ID from a S3 bucket, what not do the same thing here?

**Comment by alanshaw** [comment]
> We're going to make it consistent and move the current peerId file to the Dynamo config table.

ACK, it's the introduction of _another_ method of fetching config that's concerning for me, if we consolidate to one method then that SGTM.

> About refreshing on updating values: I didn't because looks like the method `peerStore.tagPeer` only allow to add new tagged peers, isn't it? For the `peerId`, we need to restart the service anyway too, is it right? So I haven't gone for that optimization.

My point is that updating a row in Dynamo will _not_ cause the service to restart, but the service needs to restart to pick up and tag new peers. Are we introducing a manual step where we have to restart the service?

Conversely I assume that setting an environment variable e.g. `GATEWAY_PEERS_FILE=gateway-peers-2022-12-01.json` does cause the service to restart _automatically_. Is that correct?

The other concern is how/where do we store this config value outside of Dynamo? Personally I'd keep it in the infra as code repo so that changes can be tracked and we have a backup.

**Review by alanshaw — APPROVED** [review]
Looks good to me, barring concerns with Dynamo config - IMO it would be better to stick to S3 bucket config.

File: src/config.js [hunk 1]
```diff
@@ -17,20 +17,21 @@ export function makeConfig () {
...
     dynamoRegion: process.env.DYNAMO_REGION ?? process.env.AWS_REGION,
+    configTable: process.env.DYNAMO_CONFIG_TABLE ?? 'config',
     blocksTable: process.env.DYNAMO_BLOCKS_TABLE ?? 'blocks',
     carsTable: process.env.DYNAMO_CARS_TABLE ?? 'cars',
     blocksTableV1: process.env.DYNAMO_BLOCKS_TABLE_V1 ?? 'v1-blocks',
     carsTableV1: process.env.DYNAMO_CARS_TABLE_V1 ?? 'v1-cars',
     linkTableV1: process.env.DYNAMO_LINK_TABLE_V1 ?? 'v1-blocks-cars-position',
 
+    dynamoConfigTableKey: 'key',
+    dynamoConfigTableTaggedPeersKey: 'tagged-peers',
```
[the remark this answers is not in the document]
Comment (simone-sanfratello) [reply src/config.js:28]: Yes, I validate each entry here https://github.com/elastic-ipfs/bitswap-peer/pull/170/files#diff-145d31b275fbdb6a7b17ef982f83babd2323896502f2c8374da313a8833e0ebaR17

**Comment by simone-sanfratello** [comment]
> My point is that updating a row in Dynamo will _not_ cause the service to restart, but the service needs to restart to pick up and tag new peers. Are we introducing a manual step where we have to restart the service?
> 
> Conversely I assume that setting an environment variable e.g. `GATEWAY_PEERS_FILE=gateway-peers-2022-12-01.json` does cause the service to restart _automatically_. Is that correct?
> 
> The other concern is how/where do we store this config value outside of Dynamo? Personally I'd keep it in the infra as code repo so that changes can be tracked and we have a backup.

I'd go one step at time. We can deliver that and if it fixes the performance issue, then we can talk how to refresh those values, how often they change and so on - of course a manual solution is to avoid.

File: src/index.js [hunk 2]
```diff
@@ -36,6 +36,11 @@ async function boot () {
       keyName: readinessConfig.dynamo.keyName,
       keyValue: readinessConfig.dynamo.keyValue
     })
+    const taggedPeers = await awsClient.dynamoGetItem({
```
Comment (codeflyer) [inline src/index.js:39]: What happen if doesn't find the peers?

Comment (simone-sanfratello) [reply src/index.js:39]: If the query fails, the service won't start
If the taggedPeers are empty, nothing

File: src/service.js [hunk 3]
```diff
@@ -5,19 +5,53 @@ import { noise } from '@chainsafe/libp2p-noise'
...
 import { logger as defaultLogger } from './logging.js'
+import { createPeerIdFromMultihash } from './peer-id.js'
 import inspect from './inspect/index.js'
 
-async function startService ({ peerId, port, peerAnnounceAddr, awsClient, connectionConfig, logger = defaultLogger } = {}) {
-  try {
-    // TODO params validation
+// TODO validate all the params
+function validateParams ({ taggedPeers, logger }) {
+  if (!taggedPeers) {
```
Comment (codeflyer) [inline src/service.js:18]: Does it worth to log a warning if taggedPeers are not available?

**Pull request merged**